rest/middleware.go

```diff
@@ -39,78 +46,105 @@ func (w *loggingResponseWriter) Content() string {
 	return w.buf.String()
 }
 
-func RequestLogger(debug bool, logger *zap.Logger) restful.FilterFunction {
+func RequestLoggerFilter(logger *zap.SugaredLogger) restful.FilterFunction {
 	return func(req *restful.Request, resp *restful.Response, chain *restful.FilterChain) {
 		rq := req.Request
+
 		// search a better way for a unique callid
-		// perhaps a reverseproxy in front generates a unique header for som sort
+		// perhaps a reverseproxy in front generates a unique header for some sort
 		// of opentracing support?
 
 		requestID := req.HeaderParameter("X-Request-Id")
-		rqid := zap.String("rqid", requestID)
 		if requestID == "" {
-			ts := time.Now().UnixNano()
-			rqid = zap.Int64("rqid", ts)
+			requestID = uuid.New().String()
 		}
-		fields := []zap.Field{
-			rqid,
-			zap.String("remoteaddr", strings.Split(rq.RemoteAddr, ":")[0]),
-			zap.String("method", rq.Method),
-			zap.String("uri", rq.URL.RequestURI()),
-			zap.String("route", req.SelectedRoutePath()),
+
+		clone := *logger
+		requestLogger := &clone
```

Comment:
please remove, this is a massive overhead, for separating disjunctive requests, requestID is added. A Benchmark would help to see the impact.

Reply:
This was really unnecessary because `.With` already creates a clone (see source code).